src/Sarif/Visitors/SarifCurrentToVersionOneVisitor.cs

```diff
@@ -377,38 +378,313 @@ internal PhysicalLocationVersionOne CreatePhysicalLocation(Region v2Region)
             {
                 physicalLocation = new PhysicalLocationVersionOne
                 {
-                    Region = CreateRegion(v2Region)
+                    Region = CreateRegion(v2Region, null)
                 };
             }
 
             return physicalLocation;
         }
 
-        internal RegionVersionOne CreateRegion(Region v2Region)
+        internal RegionVersionOne CreateRegion(Region v2Region, Uri uri)
         {
             RegionVersionOne region = null;
 
-            if (v2Region != null && (v2Region.StartColumn > 0 ||
-                                     v2Region.StartLine > 0 ||
-                                     v2Region.EndColumn > 0 ||
-                                     v2Region.EndLine > 0 ||
-                                     v2Region.Length > 0 ||
-                                     v2Region.Offset > 0))
-            {
-                region = new RegionVersionOne
-                {
-                    EndColumn = v2Region.EndColumn,
-                    EndLine = v2Region.EndLine,
-                    Length = v2Region.Length,
-                    Offset = v2Region.Offset,
-                    StartColumn = v2Region.StartColumn,
-                    StartLine = v2Region.StartLine
-                };
+            if (v2Region != null)
+            {
+                region = new RegionVersionOne();
+
+                if (v2Region.StartLine > 0 ||
+                    v2Region.EndLine > 0 ||
+                    v2Region.StartColumn > 0 ||
+                    v2Region.EndColumn > 0 ||
+                    v2Region.CharOffset > 0 ||
+                    v2Region.CharLength > 0)
+                {
+                    if (v2Region.StartLine > 0)
+                    {
+                        // The start of the region is described by line/column
+                        region.StartLine = v2Region.StartLine;
+                        region.StartColumn = v2Region.StartColumn > 0
+                            ? v2Region.StartColumn
+                            : 1;
+                    }
+                    else
+                    {
+                        // The start of the region is described by character offset
+
+                        if (v2Region.CharOffset == 0)
+                        {
+                            region.Offset = 0;
+                        }
+                        else
+                        {
+                            // Try to get the byte offset using the file encoding and contents
+                            region.Offset = GetRegionByteOffset(v2Region, uri);
+                        }
+                    }
+
+                    if (v2Region.CharLength > 0)
+                    {
+                        // The end of the region is described by character length
+                        // Try to get the byte length using the file encoding and contents
+                        region.Length = GetRegionByteLength(v2Region, uri);
```

Comment:
Ok, in this case your method needs both an offset and a length. So you _could_ call it `ConvertCharOffsetAndLengthToByteLength(int charOffset, int charLength, Uri uri)`. But perhaps you think that's just too long a name, and you want to keep `GetRegionByteLength`. In that case, you might argue that for symmetry, you should keep the name `GetRegionByteOffset` for the other method.

Your call.

Reply:
#NamesAreMyPassion

Reply:
Yeah but I need the start props here too

Reply:
Thanks for offline explanation. Yes, this method should stay as `GetRegionByteLength`.